core/CMakeLists.txt

```diff
@@ -43,21 +44,40 @@ if (APPLE)
             src/main/c/osx/accept.c
     )
 elseif (UNIX)
-    include_directories($ENV{JAVA_HOME}/include/linux/)
-    message(${CMAKE_SYSTEM_PROCESSOR})
-    if (CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64")
-        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/armlinux)
-        set(AARCH64 1)
-    else ()
-        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/linux)
-    endif ()
-    set(
+    EXEC_PROGRAM(uname OUTPUT_VARIABLE SYSTEM_NAME)
+    SET(SYSTEM_NAME "${SYSTEM_NAME}" CACHE INTERNAL "")
+
+    if (SYSTEM_NAME STREQUAL "FreeBSD")
+        MESSAGE("Building for FreeBSD")
+        include_directories($ENV{JAVA_HOME}/include/freebsd/)
+        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/freebsd)
+        set(
+            SOURCE_FILES ${SOURCE_FILES}
+            src/main/c/freebsd/kqueue.c
+            src/main/c/share/net.c
+            src/main/c/freebsd/affinity.c
+            src/main/c/freebsd/accept.c
+        )
+    else (SYSTEM_NAME STREQUAL "FreeBSD")
+        MESSAGE("Building for GNU/Linux")
+        include_directories($ENV{JAVA_HOME}/include/linux/)
+        message(${CMAKE_SYSTEM_PROCESSOR})
+        if (CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64")
```

Comment:
it is bit trick to change CMAKE build. Post change it has to be tested across multiple architectures. I anticipate that build will fail on ARM64 because check is done inside of FreeBSD branch - ARM64 is a linux.

Reply:
I think you have misread it, the check is done in the linux branch